### ISSUE TITLE
Export * from each module in index.ts

### ISSUE THREAD

**Issue by jdjkelly** [flexpa/sero#50]
## Goal
- [x] Use export * from path and update any examples

This is somewhat of a compromise.

### Background
tldr;

| | Native ESM Modules | TypeScript compiler support? | Submodule-style path imports |
|-|------------------------|-------------------------------|-----------------------|
| package.json#export | ✓ | X | ✓
| export * from path | ✓ | ✓ | X 

I had committed an exports configuration to package.json that was designed to deliver this:
```typescript
import { CDSHooks } from "@sero.run/sero"
import { Service } from "@sero.run/sero/cds-hooks"
```

The idea is that the [exports](https://nodejs.org/api/packages.html#packages_exports) key can be used to define additional entry points into the module beyond what is defined by "main".
```javascript
  "exports": {
    "./cds-hooks": "./dist/cds-hooks/index.js",
    "./http": "./dist/http/index.js",
    "./rest": "./dist/rest/index.js"
  }
```
It turns out that even though package.json#exports landed in Node LTS, [TypeScript compiler still doesn't support it.](https://github.com/microsoft/TypeScript/issues/33079). When it does, this should absolutely be the approach we take.

So, instead for 0.1.0 we are going to have this:
```typescript
import { CDSHooks } from "@sero.run/sero"
const { Service } from CDSHooks
```

This involves exporting _all_ of the module instead of just the default function (current behaviour) for each of the submodules (CDSHooks, Http, Rest etc). This means we won't have nice submodule-style path imports but I think that's okay for now. The alternative is adding _another_ compiler to do a pass on the project like RollUp or Webpack - and there's a lot of long term complexity that I think is better to avoid as long as possible.

**Comment by jdjkelly** [comment]
Progressed somewhat on an intermediate fix via #51 

Also need to reference https://github.com/microsoft/TypeScript/issues/33079#issuecomment-702617758
